extractCSS.js

```diff
@@ -42,7 +56,7 @@
 		var host = global.location.protocol + "//" + global.location.host;
 
 		mediaStylesheets.forEach(function (stylesheet) {
-			if ((stylesheet.href && stylesheet.href.indexOf(host) === 0) || (stylesheet.href && allowCrossDomain)) {
+			if (stylesheet.href) {
 				fetchStylesheet(stylesheet.href, function (text) {
 					var index = left.indexOf(stylesheet);
 					if (index > -1) {
```

Comment:
The weird thing about stylesheets is that while they can be loaded from an external domain just fine - accessing the raw uninterpreted CSS is a PITA.
Accessing the rules via the DOM only gives you the interpreted rules (unsupported selectors/properties/values are left out - e.g. `-ms` browser prefix gets nixed in chrome) - so that won't cut it.
Re-fetching them via XHR will - by default - get blocked by browser security... _but_ luckily PhantomJS had a way to disable the security: https://github.com/drdk/dr-css-inliner/blob/master/index.js#L217

It looks like Puppeteer has a way to do it too: the launch option `args` with the [--allow-insecure-localhost](https://peter.sh/experiments/chromium-command-line-switches/#allow-insecure-localhost) flag!

I guess `cssLinksStillLoading` was for handling fetches that failed?

Reply:
Right. I think the best way to do it with puppeteer is `page.setBypassCSP(true)`

cssLinksStillLoading is for async css links that are not loaded yet, for example `<link rel="preload" href="layout.min.css as="style" onload="this.onload=null;this.rel='stylesheet'">`

Reply:
I'm not entirely sure what `setBypassCSP` does; does it just ignore any CSP headers sent from the server (which will have no effect if none are configured?) - or does it encompass all security features of the browser (including default CORS behaviour)?

`rel="preload"` would IMHO fall outside the "critical path" that dr-css-inliner (initially - anyways) tries to cover. _But_ it could well be a feature behind a flag... (yes, yes - I love flags 😄)

Reply:
Upon further testing fetching cross-origin stylesheets via XHR doesn't seem to be a problem in chrome (or puppeteer) any more(?!)... So I guess no need for `page.setBypassCSP(true)` or `--allow-insecure-localhost` :)

Reply:
I have added "'--disable-web-security" to be as close to the behavior as it was with phantomJS.